Firestore/core/src/firebase/firestore/model/transform_operations.h

```diff
@@ -233,6 +245,116 @@ class ArrayTransform : public TransformOperation {
   }
 };
 
+/**
+ * Implements the backend semantics for locally computed NUMERIC_ADD (increment)
+ * transforms. Converts all field values to longs or doubles and resolves
+ * overflows to LONG_MAX/LONG_MIN.
+ */
+class NumericIncrementTransform : public TransformOperation {
+ public:
+  NumericIncrementTransform(FSTNumberValue* operand)
+      : operand_(std::move(operand)) {
+  }
+
+  ~NumericIncrementTransform() override {
```

Comment:
Nit: defining an empty destructor shouldn't be necessary (unless the compiler complains, in which case can you copy the warning message?).